tang.go

```diff
@@ -20,19 +20,44 @@ import (
 	"github.com/lestrrat-go/jwx/jws"
 )
 
-// DecryptTang decrypts a jwe message bound with Tang clevis pin
-func DecryptTang(msg *jwe.Message, clevisNode map[string]interface{}) ([]byte, error) {
-	tangNode, ok := clevisNode["tang"].(map[string]interface{})
-	if !ok {
-		return nil, fmt.Errorf("cannot parse provided token, node 'clevis.tang'")
+// TangPin  represents the data tang needs to perform decryption
+type TangPin struct {
```

Comment:
Some structs (like `*Pin` and `*Config`) are uppercase and they become public. Do we really need to expose it to users? If not then it should be private.

Reply:
I was thinking they should be public, since I can envision how I'd want to use this library, to interoperate with the `*Pin` and `*Config` structures in code.

Reply:
That said, there may be some of the functions I introduced as public that don't need to be...  I'll take a pass at cleaning that up a bit as I work on the unit tests.

Reply:
I have moved a few functions to be non-Public.